controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -412,10 +412,9 @@ func validateCoreDNSImageTag(fromTag, toTag string) error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to parse CoreDNS target version %q", toTag)
 	}
-	// make sure that the version we're upgrading to is greater than the current one,
-	// or if they're the same version, the raw tags should be different (e.g. allow from `v1.17.4-somevendor.0` to `v1.17.4-somevendor.1`).
-	if x := from.Compare(to); x > 0 || (x == 0 && fromTag == toTag) {
-		return fmt.Errorf("toVersion %q must be greater than fromVersion %q", toTag, fromTag)
+	// make sure that the version we're upgrading to is greater than or equal to the current version.
+	if to.LT(from) {
```

Comment:
TBH I'm not sure that we need this change given that this func is not called if version are equal https://github.com/kubernetes-sigs/cluster-api/blob/5b9922a817cfab8c8b9125a5a035d81702361e6a/controlplane/kubeadm/internal/workload_cluster_coredns.go#L128-L131